i_1006 Adjust JUnits for new TestDataGroups

johnbrvc opened the pull request:
### Description of what the PR does

1. The **JUnit** tests for the `ContestSnakeYamlLoader `made some assumptions about the locations of the secret data.  This is now handled automatically by the `TestDataGroup `class.  Fix the routine in `ContestSnakeYamlLoader `to not put the "`secret`" folder on the directory path for data.
2. Fix spelling error in **JUnit** print message.
3. Make sure the static log is created before running junits since the TestDataGroup stuff will log things.

### Issue which the PR addresses
Partially implements #1006 

### Environment in which the PR was developed (OS,IDE, Java version, etc.)
WIndows 11
java version "1.8.0_321"
Java(TM) SE Runtime Environment (build 1.8.0_321-b07)
Java HotSpot(TM) 64-Bit Server VM (build 25.321-b07, mixed mode)

### Precise steps for _testing_ the PR (i.e., how to demonstrate that it works correctly)

1. Build this PR.
2. In the Eclipse explorer, navigate to the `test.edu.csus.ecs.pc2.imports.ccs/ContestSnakeYAMLLoaderTest.java` module
3. Right click and select **Run As->JUnit Test**
4. All the tests should pass.

johnbrvc committed:
i_1006 Adjust JUnits for new TestDataGroups

The JUnit tests for the ContestSnakeYamlLoader made some assumptions about the locations of the secret data.  This is now handled automatically by the TestDataGroup class.  Fix the routine in ContestSnakeYamlLoader to not put the "secret" folder on the directory path for data.
Fix spelling error in JUnit print message.
Make sure the static log is created before running junits since the TestDataGroup stuff will log things.

clevengr reviewed (commented):
I reviewed all the changed files; the changes make sense.  I went to my clone of your fork, went onto the `i1006` branch, and verified that the `ContestSnakeYAMLLoaderTest` JUnit failed before the PR was applied.  I then pulled from your fork, switched to the `i1006_scoring_algorithm_changes` branch, rebuilt the system, and re-executed the same JUnit; all tests in that JUnit passed. 

However, I should note that when I run _ALL_ JUnit tests on that branch, there are two failures: one in `services.web.EventFeedFilterTest.testNoFilter`, the other in `services.web.EventFeedLogTest.testWriteReadCollections`.  In addition, the build process (from the PR submission) on GitHub has failures.  I'm not sure whether these are related to the PR or due to something else, so I'm reluctant to approve the PR until we know more about what's going on there...

johnbrvc committed:
i1006 Fix JUnits to include ensureStaticLog()

Some JUnit tests read a CDP that contains problem datafiles.  These JUnits now make use of the newer TestDataGroups() class to read test data as a result of using ContestSnakeYamlLoader.  This class requires a valid Log object.  ContestSnakeYamlLoader uses StaticLog(), as a result, the static log must have been previously intialized.  AbstractTestCase.ensureStaticLog() does this.

johnbrvc committed:
i_1006 Fix JUnit issues

Forgot to add an ensureStaticLog().
Also added missing call to base class setupUp() in LoadICPCDataTest.

clevengr reviewed (commented):
I reviewed all the new changed files and see no problem with them.  However, when I pull the latest updates into my clone of your fork I still get failures in JUnits.  Specifically:

```
junit.framework.AssertionFailedError: Expected line count  expected:<7> but was:<140>
	at junit.framework.Assert.fail(Assert.java:57)
	at junit.framework.Assert.failNotEquals(Assert.java:329)
	at junit.framework.Assert.assertEquals(Assert.java:78)
	at junit.framework.Assert.assertEquals(Assert.java:234)
	at junit.framework.TestCase.assertEquals(TestCase.java:401)
	at edu.csus.ecs.pc2.services.web.EventFeedFilterTest.testNoFilter(EventFeedFilterTest.java:34)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at junit.framework.TestCase.runTest(TestCase.java:176)
	at junit.framework.TestCase.runBare(TestCase.java:141)
	at junit.framework.TestResult$1.protect(TestResult.java:122)
	at junit.framework.TestResult.runProtected(TestResult.java:142)
	at junit.framework.TestResult.run(TestResult.java:125)
	at junit.framework.TestCase.run(TestCase.java:129)
	at junit.framework.TestSuite.runTest(TestSuite.java:255)
	at junit.framework.TestSuite.run(TestSuite.java:250)
	at org.junit.internal.runners.JUnit38ClassRunner.run(JUnit38ClassRunner.java:84)
	at org.eclipse.jdt.internal.junit4.runner.JUnit4TestReference.run(JUnit4TestReference.java:89)
	at org.eclipse.jdt.internal.junit.runner.TestExecution.run(TestExecution.java:41)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:542)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:770)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.run(RemoteTestRunner.java:464)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.main(RemoteTestRunner.java:210)
```
and 
```
junit.framework.AssertionFailedError: expected:<7> but was:<140>
	at junit.framework.Assert.fail(Assert.java:57)
	at junit.framework.Assert.failNotEquals(Assert.java:329)
	at junit.framework.Assert.assertEquals(Assert.java:78)
	at junit.framework.Assert.assertEquals(Assert.java:234)
	at junit.framework.Assert.assertEquals(Assert.java:241)
	at junit.framework.TestCase.assertEquals(TestCase.java:409)
	at edu.csus.ecs.pc2.services.web.EventFeedLogTest.testWriteReadCollections(EventFeedLogTest.java:69)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at junit.framework.TestCase.runTest(TestCase.java:176)
	at junit.framework.TestCase.runBare(TestCase.java:141)
	at junit.framework.TestResult$1.protect(TestResult.java:122)
	at junit.framework.TestResult.runProtected(TestResult.java:142)
	at junit.framework.TestResult.run(TestResult.java:125)
	at junit.framework.TestCase.run(TestCase.java:129)
	at junit.framework.TestSuite.runTest(TestSuite.java:255)
	at junit.framework.TestSuite.run(TestSuite.java:250)
	at org.junit.internal.runners.JUnit38ClassRunner.run(JUnit38ClassRunner.java:84)
	at org.eclipse.jdt.internal.junit4.runner.JUnit4TestReference.run(JUnit4TestReference.java:89)
	at org.eclipse.jdt.internal.junit.runner.TestExecution.run(TestExecution.java:41)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:542)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:770)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.run(RemoteTestRunner.java:464)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.main(RemoteTestRunner.java:210)

```
Are these being caused by the changes associated with this PR?

(P.S. I did verify that I have commits 1d6262c9afbe1ab8d942aa81dba98bd3af955353 and 5717d21b0b2adecbce1443e56f29b226cfbd451a, which are the two latest commits in your branch, pulled into my clone of your fork...)

johnbrvc commented:
> I reviewed all the new changed files and see no problem with them. However, when I pull the latest updates into my clone of your fork I still get failures in JUnits. Specifically:
> 
> ```
> junit.framework.AssertionFailedError: Expected line count  expected:<7> but was:<140>
> 	at junit.framework.Assert.fail(Assert.java:57)
> 	at junit.framework.Assert.failNotEquals(Assert.java:329)
> 	at junit.framework.Assert.assertEquals(Assert.java:78)
> 	at junit.framework.Assert.assertEquals(Assert.java:234)
> 	at junit.framework.TestCase.assertEquals(TestCase.java:401)
> 	at edu.csus.ecs.pc2.services.web.EventFeedFilterTest.testNoFilter(EventFeedFilterTest.java:34)
> 	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
> 	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
> 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
> 	at java.lang.reflect.Method.invoke(Method.java:498)
> 	at junit.framework.TestCase.runTest(TestCase.java:176)
> 	at junit.framework.TestCase.runBare(TestCase.java:141)
> 	at junit.framework.TestResult$1.protect(TestResult.java:122)
> 	at junit.framework.TestResult.runProtected(TestResult.java:142)
> 	at junit.framework.TestResult.run(TestResult.java:125)
> 	at junit.framework.TestCase.run(TestCase.java:129)
> 	at junit.framework.TestSuite.runTest(TestSuite.java:255)
> 	at junit.framework.TestSuite.run(TestSuite.java:250)
> 	at org.junit.internal.runners.JUnit38ClassRunner.run(JUnit38ClassRunner.java:84)
> 	at org.eclipse.jdt.internal.junit4.runner.JUnit4TestReference.run(JUnit4TestReference.java:89)
> 	at org.eclipse.jdt.internal.junit.runner.TestExecution.run(TestExecution.java:41)
> 	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:542)
> 	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:770)
> 	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.run(RemoteTestRunner.java:464)
> 	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.main(RemoteTestRunner.java:210)
> ```
> 
> and
> 
> ```
> junit.framework.AssertionFailedError: expected:<7> but was:<140>
> 	at junit.framework.Assert.fail(Assert.java:57)
> 	at junit.framework.Assert.failNotEquals(Assert.java:329)
> 	at junit.framework.Assert.assertEquals(Assert.java:78)
> 	at junit.framework.Assert.assertEquals(Assert.java:234)
> 	at junit.framework.Assert.assertEquals(Assert.java:241)
> 	at junit.framework.TestCase.assertEquals(TestCase.java:409)
> 	at edu.csus.ecs.pc2.services.web.EventFeedLogTest.testWriteReadCollections(EventFeedLogTest.java:69)
> 	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
> 	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
> 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
> 	at java.lang.reflect.Method.invoke(Method.java:498)
> 	at junit.framework.TestCase.runTest(TestCase.java:176)
> 	at junit.framework.TestCase.runBare(TestCase.java:141)
> 	at junit.framework.TestResult$1.protect(TestResult.java:122)
> 	at junit.framework.TestResult.runProtected(TestResult.java:142)
> 	at junit.framework.TestResult.run(TestResult.java:125)
> 	at junit.framework.TestCase.run(TestCase.java:129)
> 	at junit.framework.TestSuite.runTest(TestSuite.java:255)
> 	at junit.framework.TestSuite.run(TestSuite.java:250)
> 	at org.junit.internal.runners.JUnit38ClassRunner.run(JUnit38ClassRunner.java:84)
> 	at org.eclipse.jdt.internal.junit4.runner.JUnit4TestReference.run(JUnit4TestReference.java:89)
> 	at org.eclipse.jdt.internal.junit.runner.TestExecution.run(TestExecution.java:41)
> 	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:542)
> 	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:770)
> 	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.run(RemoteTestRunner.java:464)
> 	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.main(RemoteTestRunner.java:210)
> ```
> 
> Are these being caused by the changes associated with this PR?
> 
> (P.S. I did verify that I have commits [1d6262c](https://github.com/pc2ccs/pc2v9/commit/1d6262c9afbe1ab8d942aa81dba98bd3af955353) and [5717d21](https://github.com/pc2ccs/pc2v9/commit/5717d21b0b2adecbce1443e56f29b226cfbd451a), which are the two latest commits in your branch, pulled into my clone of your fork...)

I believe these go back to [this issue from a couple of years ago](https://github.com/pc2ccs/pc2v9/issues/888).

clevengr commented:
> > I reviewed all the new changed files and see no problem with them. However, when I pull the latest updates into my clone of your fork I still get failures in JUnits. 
...
> 
> I believe these go back to [this issue from a couple of years ago](https://github.com/pc2ccs/pc2v9/issues/888).

Ok, I'll buy that these are tied up in Issue #888 somehow and are not part of this PR.  I just added a comment to #888 related to this.  I'm going to approve the PR.

clevengr reviewed (approved):
The only issue I see is the failing JUnits, which seems to be unrelated to this PR (see Issue #888).  I approve the PR.
